layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -1350,10 +1350,43 @@ the right."
   (interactive)
   (set-buffer-file-coding-system 'undecided-dos nil))
 
-(defun spacemacs/copy-file ()
-  "Write the file under new name."
-  (interactive)
-  (call-interactively 'write-file))
+(defun spacemacs/copy-file (arg)
+  "Write the file under new name and select action on new file.
+This command prompts for a new filename then prompts for
+selecting an action to perform on the new file.
+
+If prefixed with the universal ARG \\[universal-argument] the
+prompt includes the buffer file name.
+
+When prompted for selecting an action, pressing `?' will open the
+help menu that explains the different options."
+  (interactive "P")
+  (let* ((new-file-name (read-file-name "Save file as: "
+                                        (if arg buffer-file-name default-directory))))
+    ;; when selecting a directory name, add (buffer-name) or message error for
+    ;; non-visiting file buffers
+    (when (directory-name-p new-file-name)
+      (if buffer-file-name
+          (setq new-file-name (concat new-file-name (buffer-name)))
+        (error "Copying of non-visiting file buffers requires to specify new file name.")))
+    (when (and (file-exists-p new-file-name)
+               (not (y-or-n-p (format "File %s exists. Overwrite?: " new-file-name))))
+      (error "Canceled"))
+    (pcase
+        (read-answer "New file action: "
+                     '(("replace"    ?r "open new file in current buffer")
+                       ("new-buffer" ?n "open new file in new buffer")
+                       ("only-write" ?o "write new-file only. Don't open")
+                       ("quit" ?q "exit")))
+      ("quit" (print "Canceled"))
+      (code (let ((dir (file-name-directory new-file-name)))
+              (when (not (file-exists-p dir))
+                (make-directory dir t)))
+            (if (string= "replace" code)
+                (write-file new-file-name))
+            (write-region (point-min) (point-max) new-file-name)
```

Comment:
write-region after write-file?

not indented correctly

Reply:
Good catch! And thanks for your review :+1: